plugins/flatpak/gs-flatpak.c

```diff
@@ -2617,6 +2664,22 @@ gs_flatpak_refine_runtime_for_install (GsFlatpak *self,
 	return TRUE;
 }
 
+static void
+gs_flatpak_add_app_to_list_if_not_installed (GsApp *app, GsAppList *list,
+					     const GHashTable *hash_installed)
+{
+	g_autofree gchar *ref_display = gs_flatpak_app_get_ref_display (app);
+	if (g_hash_table_contains (hash_installed, ref_display)) {
+		g_debug ("%s is already installed, so skipping",
+			 gs_app_get_id (app));
```

Comment:
You could do an early return here to skip the else block.

Reply:
Fixed.